diff -ur qtwebengine-opensource-src-5.8.0/src/3rdparty/chromium/third_party/webrtc/system_wrappers/BUILD.gn qtwebengine-opensource-src-5.8.0-webrtc-neon-detect/src/3rdparty/chromium/third_party/webrtc/system_wrappers/BUILD.gn
--- qtwebengine-opensource-src-5.8.0/src/3rdparty/chromium/third_party/webrtc/system_wrappers/BUILD.gn	2017-01-03 10:28:53.000000000 +0100
+++ qtwebengine-opensource-src-5.8.0-webrtc-neon-detect/src/3rdparty/chromium/third_party/webrtc/system_wrappers/BUILD.gn	2017-03-05 17:08:43.547748044 +0100
@@ -108,9 +108,7 @@
   if (is_linux) {
     defines += [ "WEBRTC_THREAD_RR" ]
 
-    if (!build_with_chromium) {
-      deps += [ ":cpu_features_linux" ]
-    }
+    deps += [ ":cpu_features_linux" ]
 
     libs += [ "rt" ]
   }
diff -ur qtwebengine-opensource-src-5.8.0/src/3rdparty/chromium/third_party/webrtc/system_wrappers/system_wrappers.gyp qtwebengine-opensource-src-5.8.0-webrtc-neon-detect/src/3rdparty/chromium/third_party/webrtc/system_wrappers/system_wrappers.gyp
--- qtwebengine-opensource-src-5.8.0/src/3rdparty/chromium/third_party/webrtc/system_wrappers/system_wrappers.gyp	2017-01-03 10:28:53.000000000 +0100
+++ qtwebengine-opensource-src-5.8.0-webrtc-neon-detect/src/3rdparty/chromium/third_party/webrtc/system_wrappers/system_wrappers.gyp	2017-03-05 17:07:01.869254856 +0100
@@ -112,12 +112,8 @@
           'defines': [
             'WEBRTC_THREAD_RR',
           ],
-          'conditions': [
-            ['build_with_chromium==0', {
-              'dependencies': [
-                'cpu_features_webrtc.gyp:cpu_features_linux',
-              ],
-            }],
+          'dependencies': [
+            'cpu_features_webrtc.gyp:cpu_features_linux',
           ],
           'link_settings': {
             'libraries': [ '-lrt', ],
