diff --git a/src/3rdparty/chromium/base/debug/profiler.h b/src/3rdparty/chromium/base/debug/profiler.h
index be6e272e0..b27f5f690 100644
--- a/src/3rdparty/chromium/base/debug/profiler.h
+++ b/src/3rdparty/chromium/base/debug/profiler.h
@@ -5,6 +5,7 @@
 #ifndef BASE_DEBUG_PROFILER_H_
 #define BASE_DEBUG_PROFILER_H_

+#include <cstdint>
 #include <stddef.h>
 #include <cstdint>

diff --git a/src/3rdparty/chromium/cc/trees/target_property.cc b/src/3rdparty/chromium/cc/trees/target_property.cc
index 00595fda0..ebd8c605a 100644
--- a/src/3rdparty/chromium/cc/trees/target_property.cc
+++ b/src/3rdparty/chromium/cc/trees/target_property.cc
@@ -6,6 +6,10 @@

 #include "cc/trees/target_property.h"

+#include <cstdint>
+
+#include <cstdint>
+
 namespace cc {

 static_assert(TargetProperty::LAST_TARGET_PROPERTY < kMaxTargetPropertyIndex,
diff --git a/src/3rdparty/chromium/device/base/synchronization/one_writer_seqlock.cc b/src/3rdparty/chromium/device/base/synchronization/one_writer_seqlock.cc
index 98266a16c..7676e6097 100644
--- a/src/3rdparty/chromium/device/base/synchronization/one_writer_seqlock.cc
+++ b/src/3rdparty/chromium/device/base/synchronization/one_writer_seqlock.cc
@@ -6,6 +6,8 @@

 #include "device/base/synchronization/one_writer_seqlock.h"

+#include <cstdint>
+
 namespace device {

 OneWriterSeqLock::OneWriterSeqLock() : sequence_(0) {}
diff --git a/src/3rdparty/chromium/extensions/common/constants.h b/src/3rdparty/chromium/extensions/common/constants.h
index e6183dc11..5a12758ac 100644
--- a/src/3rdparty/chromium/extensions/common/constants.h
+++ b/src/3rdparty/chromium/extensions/common/constants.h
@@ -13,6 +13,8 @@
 #include "components/version_info/channel.h"
 #include "ui/base/layout.h"

+#include <cstdint>
+
 namespace extensions {

 // Scheme we serve extension content from.
diff --git a/src/3rdparty/chromium/gpu/config/gpu_preferences.h b/src/3rdparty/chromium/gpu/config/gpu_preferences.h
index 57bda1f99..7a5ef9b3a 100644
--- a/src/3rdparty/chromium/gpu/config/gpu_preferences.h
+++ b/src/3rdparty/chromium/gpu/config/gpu_preferences.h
@@ -5,6 +5,7 @@
 #ifndef GPU_CONFIG_GPU_PREFERENCES_H_
 #define GPU_CONFIG_GPU_PREFERENCES_H_

+#include <cstdint>
 #include <stddef.h>
 #include <cstdint>
 #include <string>
diff --git a/src/3rdparty/chromium/net/base/parse_number.h b/src/3rdparty/chromium/net/base/parse_number.h
index 301d14ac8..378058a23 100644
--- a/src/3rdparty/chromium/net/base/parse_number.h
+++ b/src/3rdparty/chromium/net/base/parse_number.h
@@ -11,6 +11,8 @@
 #include "base/strings/string_piece.h"
 #include "net/base/net_export.h"

+#include <cstdint>
+
 // This file contains utility functions for parsing numbers, in the context of
 // network protocols.
 //
diff --git a/src/3rdparty/chromium/pdf/document_attachment_info.h b/src/3rdparty/chromium/pdf/document_attachment_info.h
index fd78cb729..f9330a0db 100644
--- a/src/3rdparty/chromium/pdf/document_attachment_info.h
+++ b/src/3rdparty/chromium/pdf/document_attachment_info.h
@@ -9,6 +9,8 @@

 #include "base/strings/string16.h"

+#include <cstdint>
+
 namespace chrome_pdf {

 struct DocumentAttachmentInfo {
diff --git a/src/3rdparty/chromium/ppapi/utility/completion_callback_factory_thread_traits.h b/src/3rdparty/chromium/ppapi/utility/completion_callback_factory_thread_traits.h
index ac0b31a75..5d1c421d0 100644
--- a/src/3rdparty/chromium/ppapi/utility/completion_callback_factory_thread_traits.h
+++ b/src/3rdparty/chromium/ppapi/utility/completion_callback_factory_thread_traits.h
@@ -38,6 +38,10 @@ namespace pp {
 /// As a further optimization, we can add support for this later.
 class ThreadSafeThreadTraits {
  public:
+
+  typedef pp::Lock Lock;
+  typedef pp::AutoLock AutoLock;
+
   class RefCount {
    public:
     /// Default constructor. In debug mode, this checks that the object is being
@@ -67,8 +71,6 @@ class ThreadSafeThreadTraits {
     int32_t ref_;
   };

-  typedef pp::Lock Lock;
-  typedef pp::AutoLock AutoLock;
 };

 /// The non-thread-safe version of thread traits. Using this class as the
diff --git a/src/3rdparty/chromium/third_party/angle/src/common/angleutils.h b/src/3rdparty/chromium/third_party/angle/src/common/angleutils.h
index fb9a44225..f9330f313 100644
--- a/src/3rdparty/chromium/third_party/angle/src/common/angleutils.h
+++ b/src/3rdparty/chromium/third_party/angle/src/common/angleutils.h
@@ -15,6 +15,7 @@
 #    include "absl/container/flat_hash_map.h"
 #endif  // defined(ANGLE_USE_ABSEIL)

+#include <cstdint>
 #include <climits>
 #include <cstdarg>
 #include <cstddef>
diff --git a/src/3rdparty/chromium/third_party/libgav1/src/src/utils/threadpool.cc b/src/3rdparty/chromium/third_party/libgav1/src/src/utils/threadpool.cc
index 4e8788bc4..a5d35b31b 100644
--- a/src/3rdparty/chromium/third_party/libgav1/src/src/utils/threadpool.cc
+++ b/src/3rdparty/chromium/third_party/libgav1/src/src/utils/threadpool.cc
@@ -31,6 +31,7 @@
 #include <cstdio>
 #include <cstdint>
 #include <cstring>
+#include <cstdio>
 #include <new>
 #include <utility>

diff --git a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/proto/proto_importer_module.h b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/proto/proto_importer_module.h
index 098433c47..0820bb220 100644
--- a/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/proto/proto_importer_module.h
+++ b/src/3rdparty/chromium/third_party/perfetto/src/trace_processor/importers/proto/proto_importer_module.h
@@ -23,6 +23,8 @@
 #include "perfetto/trace_processor/status.h"
 #include "src/trace_processor/trace_blob_view.h"

+#include <cstdint>
+
 namespace perfetto {

 namespace protos {
diff --git a/src/3rdparty/chromium/ui/events/gesture_event_details.h b/src/3rdparty/chromium/ui/events/gesture_event_details.h
index d20416738..8dadab3d0 100644
--- a/src/3rdparty/chromium/ui/events/gesture_event_details.h
+++ b/src/3rdparty/chromium/ui/events/gesture_event_details.h
@@ -5,6 +5,7 @@
 #ifndef UI_EVENTS_GESTURE_DETECTION_GESTURE_EVENT_DETAILS_H_
 #define UI_EVENTS_GESTURE_DETECTION_GESTURE_EVENT_DETAILS_H_

+#include <cstdint>
 #include <string.h>

 #include "base/check_op.h"
diff --git a/src/3rdparty/chromium/ui/gl/gl_surface_glx.cc b/src/3rdparty/chromium/ui/gl/gl_surface_glx.cc
index 1d5527691..ef9d5ce13 100644
--- a/src/3rdparty/chromium/ui/gl/gl_surface_glx.cc
+++ b/src/3rdparty/chromium/ui/gl/gl_surface_glx.cc
@@ -4,6 +4,7 @@

 #include "ui/gl/gl_surface_glx.h"

+#include <cstdint>
 #include <utility>

 #include "base/bind.h"
diff --git a/src/3rdparty/chromium/ui/gl/gl_surface_glx.h b/src/3rdparty/chromium/ui/gl/gl_surface_glx.h
index ab612cd87..2178d8514 100644
--- a/src/3rdparty/chromium/ui/gl/gl_surface_glx.h
+++ b/src/3rdparty/chromium/ui/gl/gl_surface_glx.h
@@ -5,6 +5,7 @@
 #ifndef UI_GL_GL_SURFACE_GLX_H_
 #define UI_GL_GL_SURFACE_GLX_H_

+#include <cstdint>
 #include <stdint.h>

 #include <memory>
diff --git a/src/3rdparty/chromium/v8/src/base/logging.h b/src/3rdparty/chromium/v8/src/base/logging.h
index 7c774ed30..1c44cf72a 100644
--- a/src/3rdparty/chromium/v8/src/base/logging.h
+++ b/src/3rdparty/chromium/v8/src/base/logging.h
@@ -5,6 +5,7 @@
 #ifndef V8_BASE_LOGGING_H_
 #define V8_BASE_LOGGING_H_

+#include <cstdint>
 #include <cstring>
 #include <cstdint>
 #include <sstream>
diff --git a/src/core/browsing_data_remover_delegate_qt.h b/src/core/browsing_data_remover_delegate_qt.h
index 5adfbbe7e..4e5c59e33 100644
--- a/src/core/browsing_data_remover_delegate_qt.h
+++ b/src/core/browsing_data_remover_delegate_qt.h
@@ -44,6 +44,8 @@

 #include "content/public/browser/browsing_data_remover_delegate.h"

+#include <cstdint>
+
 namespace QtWebEngineCore {

 class BrowsingDataRemoverDelegateQt : public content::BrowsingDataRemoverDelegate {
