Index: src/3rdparty/chromium/build/toolchain/linux/BUILD.gn
diff --git a/src/3rdparty/chromium/build/toolchain/linux/BUILD.gn b/src/3rdparty/chromium/build/toolchain/linux/BUILD.gn
index 548722d3938017ac936b9088b7a686c2afbbb5fe..86cd7dabc6e86d22a266817dfbb3bf0c719f7992 100644
--- a/src/3rdparty/chromium/build/toolchain/linux/BUILD.gn
+++ b/src/3rdparty/chromium/build/toolchain/linux/BUILD.gn
@@ -21,6 +21,24 @@ clang_toolchain("clang_arm64") {
   }
 }
 
+gcc_toolchain("arm64") {
+  toolprefix = "aarch64-linux-gnu-"
+
+  cc = "${toolprefix}gcc"
+  cxx = "${toolprefix}g++"
+
+  ar = "${toolprefix}ar"
+  ld = cxx
+  readelf = "${toolprefix}readelf"
+  nm = "${toolprefix}nm"
+
+  toolchain_args = {
+    current_cpu = "arm64"
+    current_os = "linux"
+    is_clang = false
+  }
+}
+
 gcc_toolchain("arm") {
   toolprefix = "arm-linux-gnueabihf-"
 
