diff -ur qtwebengine-opensource-src-5.7.1/src/3rdparty/chromium/skia/skia_library.gypi qtwebengine-opensource-src-5.7.1-skia-neon/src/3rdparty/chromium/skia/skia_library.gypi
--- qtwebengine-opensource-src-5.7.1/src/3rdparty/chromium/skia/skia_library.gypi	2016-11-07 15:46:18.000000000 +0100
+++ qtwebengine-opensource-src-5.7.1-skia-neon/src/3rdparty/chromium/skia/skia_library.gypi	2016-12-04 14:34:12.531877529 +0100
@@ -159,6 +159,11 @@
         '../build/android/ndk.gyp:cpu_features',
       ],
     }],
+    [ 'OS != "android" and target_arch == "arm"', {
+      'sources': [
+        '../third_party/skia/src/core/SkUtilsArm.cpp',
+      ],
+    }],
     [ 'desktop_linux == 1 or chromeos == 1 or qt_os=="embedded_linux"', {
       'dependencies': [
         '../build/linux/system.gyp:fontconfig',
