diff -up qtwebengine-everywhere-src-5.15.5/src/core/web_engine_library_info.cpp.no-icudtl-dat qtwebengine-everywhere-src-5.15.5/src/core/web_engine_library_info.cpp
--- qtwebengine-everywhere-src-5.15.5/src/core/web_engine_library_info.cpp.no-icudtl-dat	2021-06-24 07:26:58.976486102 -0500
+++ qtwebengine-everywhere-src-5.15.5/src/core/web_engine_library_info.cpp	2021-06-24 07:32:19.272863523 -0500
@@ -273,7 +273,6 @@ QString dictionariesPath()
 
 QString resourcesDataPath()
 {
-    static bool initialized = false;
     static QString potentialResourcesPath =
 #if defined(OS_MAC) && defined(QT_MAC_FRAMEWORK_BUILD)
             getResourcesPath(frameworkBundle());
@@ -282,21 +281,6 @@ QString resourcesDataPath()
 #else
             QLibraryInfo::location(QLibraryInfo::DataPath) % QLatin1String("/resources");
 #endif
-    if (!initialized) {
-        initialized = true;
-        if (!QFileInfo::exists(potentialResourcesPath % QLatin1String("/qtwebengine_resources.pak"))) {
-            qWarning("Qt WebEngine resources not found at %s. Trying parent directory...", qPrintable(potentialResourcesPath));
-            potentialResourcesPath = QLibraryInfo::location(QLibraryInfo::DataPath);
-        }
-        if (!QFileInfo::exists(potentialResourcesPath % QLatin1String("/qtwebengine_resources.pak"))) {
-            qWarning("Qt WebEngine resources not found at %s. Trying application directory...", qPrintable(potentialResourcesPath));
-            potentialResourcesPath = QCoreApplication::applicationDirPath();
-        }
-        if (!QFileInfo::exists(potentialResourcesPath % QLatin1String("/qtwebengine_resources.pak"))) {
-            qWarning("Qt WebEngine resources not found at %s. Trying fallback directory... The application MAY NOT work.", qPrintable(potentialResourcesPath));
-            potentialResourcesPath = fallbackDir();
-        }
-    }
 
     return potentialResourcesPath;
 }
