diff -ur qtwebengine-opensource-src-5.7.1/src/core/gyp_run.pro qtwebengine-opensource-src-5.7.1-arm-fpu-fix/src/core/gyp_run.pro
--- qtwebengine-opensource-src-5.7.1/src/core/gyp_run.pro	2016-11-09 06:28:31.000000000 +0100
+++ qtwebengine-opensource-src-5.7.1-arm-fpu-fix/src/core/gyp_run.pro	2016-12-03 16:38:33.531711958 +0100
@@ -80,7 +80,7 @@
         # we use arm_neon_optional for ARMv7 and newer and let chromium decide
         # about the mfpu option.
         contains(MFPU, ".*neon.*"): GYP_CONFIG += arm_fpu=\"$$MFPU\" arm_neon=1
-        else:!lessThan(MARMV, 7): GYP_CONFIG += arm_neon=0 arm_neon_optional=1
+        else:!lessThan(MARMV, 7): GYP_CONFIG += arm_fpu=\"$$MFPU\" arm_neon=0 arm_neon_optional=1
         else: GYP_CONFIG += arm_fpu=\"$$MFPU\" arm_neon=0 arm_neon_optional=0
     } else {
         # Chromium defaults to arm_neon=1, Qt does not.
