diff -ur qtwebengine-opensource-src-5.7.0/src/3rdparty/chromium/third_party/webrtc/modules/video_processing/video_processing.gypi qtwebengine-opensource-src-5.7.0-webrtc-neon/src/3rdparty/chromium/third_party/webrtc/modules/video_processing/video_processing.gypi
--- qtwebengine-opensource-src-5.7.0/src/3rdparty/chromium/third_party/webrtc/modules/video_processing/video_processing.gypi	2016-05-26 15:53:47.000000000 +0200
+++ qtwebengine-opensource-src-5.7.0-webrtc-neon/src/3rdparty/chromium/third_party/webrtc/modules/video_processing/video_processing.gypi	2016-07-18 14:29:24.841980563 +0200
@@ -47,7 +47,7 @@
         ['target_arch=="ia32" or target_arch=="x64"', {
           'dependencies': [ 'video_processing_sse2', ],
         }],
-        ['target_arch=="arm" or target_arch == "arm64"', {
+        ['build_with_neon==1', {
           'dependencies': [ 'video_processing_neon', ],
         }],
       ],
@@ -77,7 +77,7 @@
         },
       ],
     }],
-    ['target_arch=="arm" or target_arch == "arm64"', {
+    ['build_with_neon==1', {
       'targets': [
         {
           'target_name': 'video_processing_neon',
