diff --git a/src/buildtools/gn.pro b/src/buildtools/gn.pro
index b6bf9cfc4..30f92e6ac 100644
--- a/src/buildtools/gn.pro
+++ b/src/buildtools/gn.pro
@@ -18,7 +18,7 @@ build_pass|!debug_and_release {
             src_3rd_party_dir = $$absolute_path("$${getChromiumSrcDir()}/../", "$$QTWEBENGINE_ROOT")
             gn_bootstrap = $$system_path($$absolute_path(gn/build/gen.py, $$src_3rd_party_dir))
 
-            gn_gen_args = --no-last-commit-position --out-path $$out_path \
+            gn_gen_args = --no-last-commit-position --verbose --out-path $$out_path \
                           --cc \"$$which($$QMAKE_CC)\" --cxx \"$$which($$QMAKE_CXX)\" \
                           --ld \"$$which($$QMAKE_LINK)\"
 
